pkg/Directory.Build.props

```diff
@@ -17,7 +17,7 @@
        We'll add a RID for the current platform even if it isn't in the officially supported set -->
   <ItemGroup Condition="'@(OfficialBuildRID)' != ''">
     <BuildRID Include="@(OfficialBuildRID)" Exclude="$(PackageRID)"/>
-    <BuildRID Include="$(PackageRID)">
+    <BuildRID Include="$(PackageRID)" Condition="'$(BuildsEvenIfRIDNotOfficiallySupported)' == 'true'">
```

Comment:
That might be a problem, but the logic in `pkg/dir.traversal.targets` looks like a bigger problem:

```
%(Project.PackageTargetRuntime)$(PackageTargetRuntimeSuffixWithDash)' == '$(PackageRID)'
```

Since `PackageTargetRuntimeSuffixWithDash` is (I think) set based on the current configuration, rather than `Project`'s target, this makes the `win10-x64-aot` configuration/PackageRID match up with the `win10-x64` Project and *not* the `win10-x64-aot` Project. I believe this causes a mismatch resulting in the errors this change is hitting:

```
pkg\frameworkPackage.targets(61,5): error : IncludeLibFiles was specified but no file props were found in 'C:\git\source-build\src\corefx\artifacts\bin\pkg\uap\lib'
```

My scenario on Linux only happened to work because I never wanted an `aot` RID. Working on a better fix.